docs/terraform/tf-state.md

```diff
@@ -0,0 +1,72 @@
+# Using Object Storage as a Terraform Backend
+
+By default, Terraform stores its state on the local disk of the developer or operator. This approach limits collaboration, risks data loss, and hinders automation.
+
+[Read more about what Terraform state is and why it exists here.](https://developer.hashicorp.com/terraform/language/state)
+Terraform supports various backends that allow you to store state in external systems. In this tutorial, we'll be setting up our Terraform project to use [S3 compatible object storage](https://developer.hashicorp.com/terraform/language/backend/s3) as backend state storage.
+
+
+### 0. Requirements
+
+This tutorial expects you to have set up the openstack command line tool. See [Getting Started -> How to use the Openstack CLI](../../Getting-Started/Using-Openstack-CLI) for a detailed guide.
+
+## 1. Cloud Resources
+
+Create an object storage container to host the statefiles and ec2-compatible credentials to access this container programmatically:
+```bash
+openstack container create tf-state
+```
+```bash
+openstack ec2 credentials create
+```
+The latter command will output the access credentials and secret we'll plug into our terraform backend configuration in the next step.
+
+
+## 2. Terraform Code
+
+What follows is a bare minimum Terraform configuration to get you started on your LeafCloud. Pay special attention to the `backend` block, which is where the credentials we generated in our last step are to be plugged in.
+```C
+terraform {
+required_version = ">= 0.14.0"
+  required_providers {
+    openstack = {
+      source  = "terraform-provider-openstack/openstack"
+      version = "~> 1.53.0"
+    }
+  }
+}
+
+provider "openstack" {
+  auth_url    = "https://create.leaf.cloud:5000"
+  region      = "europe-nl"
+  use_octavia = true # support octavia loadbalancer
+  tenant_name = "cool-site-project-name"

```

Comment:
not sure of tenant_name and project_id are both required

Reply:
That's right.  Both are optional (uses ENV values if absent).  Will leave in the tenant_id to keep things clear.